compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -1497,12 +1497,18 @@ fn print_native_static_libs(
                 | NativeLibKind::Dylib { .. }
                 | NativeLibKind::Unspecified => {
                     let verbatim = lib.verbatim;
-                    if sess.target.is_like_msvc {
-                        Some(format!("{}{}", name, if verbatim { "" } else { ".lib" }))
-                    } else if sess.target.linker_flavor.is_gnu() {
-                        Some(format!("-l{}{}", if verbatim { ":" } else { "" }, name))
-                    } else {
-                        Some(format!("-l{name}"))
+                    match sess.target.linker_flavor {
+                        LinkerFlavor::Gnu(Cc::Yes, Lld::No) if verbatim => {
+                            Some(format!("-Wl,{}", name))
+                        }
+                        LinkerFlavor::Gnu(Cc::No, Lld::No) if verbatim => Some(name.to_string()),
+                        LinkerFlavor::Gnu(..) => {
+                            Some(format!("-l{}{}", if verbatim { ":" } else { "" }, name))
+                        }
+                        _ if sess.target.is_like_msvc => {
+                            Some(format!("{}{}", name, if verbatim { "" } else { ".lib" }))
+                        }
+                        _ => Some(format!("-l{name}")),
```

Comment:
Discussion: this seems... very tricky.

(This question is not necessarily directed at you) Are we even able to test for this in CI, as in picking *different* linkers?

Reply:
Yeah, that's what I was thinking, it's a matter of if we even have these linkers in CI installed.